### ISSUE TITLE
Oxleas Deployment: Host in SLaM Azure Cloud

### ISSUE THREAD

**Issue by MollieCourtenay** [healthlocker/oxleas-adhd#50]
**As a** product owner
**I want** the application deployed to the SLaM Azure cloud
**So that** I can use and test the application and share it with users and stakeholders.

## dwyl List of Steps:

1. #106. Spike: Host basic phoenix application on azure test environment by following the [documentation](https://github.com/dwyl/learn-phoenix-framework/blob/master/production-deployment.md). (T4h)
>This is a spike that will allow us to better understand the requirements and thus provide further time estimates. This will include reading and understanding the necessary documentation.

2. #107. Deploy basic umbrella app. (T1d)
>This will involve configuring edeliver (T2h), Setting up the azure server for deploying the app (T2h), building with edeliver (T4h)

3. #108. Deploying complex umbrella app. (~T2d)
>This cannot be thoroughly broken down or accurately estimated until 1 and 2 have been completed

**Comment by RobStallion** [comment]
Currently reading through technical requirements needed to deploy to SLaM's Azure Infrastructure [here](https://github.com/healthlocker/healthlocker/issues/841)

**Comment by RobStallion** [comment]
From what I have read it appears as though SLaM will need to provision the VM machine/machines to allow us to host in SLaM's Azure Infrastructure. @nelsonic am I understanding this correctly or did SLaM have to set it up previously to allow connection to the ePJS database?

Regarding the actual deployment to Azure, there is extensive documentation which can be found [here](https://github.com/dwyl/learn-phoenix-framework/blob/master/production-deployment.md) and contains all the information needed to deploy a basic phoenix application to Azure.

**Comment by nelsonic** [comment]
@RobStallion correct. (_SLaM have to provision new VMs_)

**Comment by iteles** [comment]
https://github.com/healthlocker/oxleas-adhd/issues/70#issuecomment-319379508
<img width="620" alt="screen shot 2017-08-01 at 15 59 04" src="https://user-images.githubusercontent.com/4185328/28831737-63a4869e-76d2-11e7-9c61-8eeeac6a27da.png">

@MollieCourtenay Thanks for this info! Could you please ask SLaM to confirm that they are able to provision both a development/test environment and a staging environment as requested?

**Comment by MollieCourtenay** [comment]
STUART needs to know:
Had no detail over backup if required, recovery time if required. The server will be provisioned the same as SLaM. Still no detail as to how the server is configured from DWYL. Where is the postgress DB? Is it in the Docker cluster? This needs documenting and handing over.

There has been no clarity as to how the server configuration has been finally setup. There is a high-level diagram but nothing with detail as what is inside the Docker container. Please remember that this Linux server was handed over to the developer DWYL as we built out with them. No detail documentation as to what has been setup inside this server has been seen only HLD.

**Comment by reddog** [comment]
As discussed elsewhere, this is blocked by https://github.com/healthlocker/healthlocker/issues/340

Please do let us know if you need any further guidance/clarity on what documentation is expected at this stage.  Also please let us know who would be able to provide the documentation and when that would be possible.

**Comment by markwilliamfirth** [comment]
@katbow @RobStallion is this one ready to be closed now?

**Comment by reddog** [comment]
Reopening issue as somewhere for remaining steps to be listed, time estimated and have someone assigned to.  Please see https://github.com/healthlocker/oxleas-adhd/issues/108#issuecomment-324381692 and other comments.

**Comment by markwilliamfirth** [comment]
@reddog this issue is for deployment of the app and this has been completed, the comment that is referenced mentions:

AWS SES setup & configuration
nginx
Travis
SSL

However it is not clear what the requirements are and there is already an issue for SSL for example ( #94 ) - please could you clarify the requirements in a new issue as this issue should be closed as it has already been completed.

**Comment by markwilliamfirth** [comment]
@reddog 

in response to:
>Have tried calling a number of times as I completely agree with your previous comments about communication efficiency, however I can't get hold of you. Re. tagging you on GitHub, please see #108 and #50. I completely agree about getting requirements clear and hence my efforts to understand what you (the development team) have those requirements listed as.

Hi David - I'm unavailable to take calls this morning (https://github.com/healthlocker/oxleas-adhd/issues/108#issuecomment-324451776)

All requirements that the dev team have are recorded on github. Please let us know any additional requirements to what we already have (such as #94 ) and we can then address these.

**Comment by reddog** [comment]
@markwilliamfirth Thanks for your comments.

Which issue is the correct issue to reference with regard to the whole project being installed? Maybe this is an epic?

Which issue shows nginx, for example, as a task with time estimates and assignees?

Can you please list:

- [ ] installation task
- [ ] issue for that task
- [ ] time estimate of that task
- [ ] assignee
- [ ] when it is scheduled for

**Comment by markwilliamfirth** [comment]
@reddog please see here: https://github.com/healthlocker/oxleas-adhd/issues/127

Please feel free to create any additional issues for information or tasks you require and I can address them

**Comment by reddog** [comment]
@markwilliamfirth Thanks for creating that new issue.

Can you please reply with which issue relates to the whole of the project being installed (ie all constituent parts)?

It is not for me to confirm if there are any other steps. I have previously commented that we have insufficient documentation to determine if all steps are covered. As such, can you please ensure that all remaining issues are listed and referenced accordingly in the *whole project installation* issue.

As a further example, which issue relates to AWS SES configuration? What is the time estimate, who is assigned, when is it scheduled for?

**Comment by markwilliamfirth** [comment]
@reddog 

>Can you please reply with which issue relates to the whole of the project being installed (ie all constituent parts)?

There is not an issue for this as far as I can see

>It is not for me to confirm if there are any other steps.

Our processes require product owners / project managers to create issues with their requirements - you don't have to list all the technical steps, just what you wish us to do

>As such, can you please ensure that all remaining issues are listed and referenced accordingly in the whole project installation issue.

Please can you further define what you mean by "whole project installation" - this is subjective and could lead to miscommunication. If possible please break down what you mean by this, the technical steps are not required, we just need to know what outcomes you want.

**Comment by markwilliamfirth** [comment]
@reddog please see here for an issue for AWS: https://github.com/healthlocker/oxleas-adhd/issues/130

Please feel free to create any additional issues describing the outcomes you want

**Comment by reddog** [comment]
@markwilliamfirth Thanks for opening the new issue on AWS SES.

As one doesn't already exist, so that we can ensure nothing falls through the cracks, can you please open an issue that pulls together the remaining setup steps.  Ideally this would have been the issue (epic?) that included all components, eg Postgres, deployment of umbrella app, nginx, etc.

Thanks very much,

**Comment by markwilliamfirth** [comment]
@reddog in order to make issues for technical steps these need to come from a requirement / outcome from the client - please could you make a "full project installation" issue defining this and describing the outcomes you wish and we can proceed from there

I have created a separate Postgres issue as you have requested this. Please feel free to create any additional issues describing the outcomes you want.

**Comment by reddog** [comment]
@markwilliamfirth Thanks for creating the Postgres issue.  Although not previously split out as a separate issue, this is one of the steps that has been completed, see https://github.com/healthlocker/oxleas-adhd/issues/110#issuecomment-323742234  As such, should it be `priority-1`?

**Comment by markwilliamfirth** [comment]
@reddog it is a priority 1 as it has been requested by yourself to be looked into with urgency

Please close it if it is not needed and please open any other additional issues and add priority labels as appropriate

As the postgres step has already been completed when Katherine and Robert see it they will update the issue to say this

**Comment by reddog** [comment]
@markwilliamfirth Thanks for confirming that.  I had mentioned the Postgres setup as an example of a *previously completed* step.

Other than lacking documentation (dealt with elsewhere), Is it useful for Katherine and Rob to have to review the newly created issue at this stage?

Thanks.

**Comment by markwilliamfirth** [comment]
@reddog I created the issue as I understood that was your request. I have now closed it based on your comment above.

As stated before - please create any issues with requirements, outcomes and priority labels that you wish me to address. This avoids any potential miscommunication. Thank you!

**Comment by reddog** [comment]
@markwilliamfirth Thanks for that.

Do we also now have an issue that pulls together of the functional components required for completion of the app installation?  For example nginx, AWS SES, etc?

Thanks again.

**Comment by markwilliamfirth** [comment]
@reddog please create any issues with requirements, outcomes and priority labels that you wish me to address. This avoids any potential miscommunication and this is the process we work by, I'm sorry that I'm unable to help otherwise, but this is the process that must be followed with regards to our development.

**Comment by reddog** [comment]
@markwilliamfirth I have created https://github.com/healthlocker/oxleas-adhd/issues/138 and I hope this fully covers anything remaining.

**Comment by markwilliamfirth** [comment]
@reddog thanks! Closing this issue (50) as it is complete
